simple_repository/components/priority_selected.py

```diff
@@ -112,15 +112,32 @@ async def get_resource(
         *,
         request_context: model.RequestContext = model.RequestContext.DEFAULT,
     ) -> model.Resource:
+        """Retrieves a resource from the first source that has the project.
+
+        This follows the same first-seen policy as get_project_page - once we find
+        the first repository that has the project page, we only look for the resource
+        in that repository.
+        """
         for source in self.sources:
             try:
-                resource = await source.get_resource(
+                # First check if this source has the project page
+                await source.get_project_page(
```

Comment:
FWIW, I don't love the fact that we have to do an additional (etag-less) request here.

At present, I think the resource handling model is wrong. #26 relates to that